### ISSUE TITLE
Invalid unsupported string offset operation

### ISSUE THREAD

**Issue by iakoder** [kalessil/phpinspectionsea#886]
![screen shot 2018-04-18 at 5 42 46 pm](https://user-images.githubusercontent.com/13456541/38965274-99164dfa-4330-11e8-81a8-5fe26264743c.png)
![screen shot 2018-04-18 at 5 46 05 pm](https://user-images.githubusercontent.com/13456541/38965275-9942a490-4330-11e8-9041-135300618256.png)

I've run into one case where I'm getting an unsupported string offset operation that, other than corrupt metadata, I can't explain.  It's assigning into a deeply nested array but the type of the variable is just array.

Yea, I tried to make a simple test case and it's fine.  I tried copying in the entire file and it's fine too.  So, no, can't get you a reproducible case.  Sorry.

PhpStorm 2018.1
Build #PS-181.4203.565, built on March 28, 2018
Licensed to Intacct / Dan Wilks
You have a perpetual fallback license for this version
Subscription is active until April 6, 2019
JRE: 1.8.0_152-release-1136-b20 x86_64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
macOS 10.13.2

Php Inspections (EA Extended) 3.0.0

No errors in the idea.log during the inspection.  Nothing in the log mentioning the class name or the field name.

**Comment by kalessil** [comment]
Difficult to say what causes this. Ideally following should help, can you check, please?
```php
    /* @var string[][][] */
    private $emailData;
```

**Comment by iakoder** [comment]
```
    /**
     * @var string[][][] $emailData
     */
    var $emailData = [];

```

PhpStorm type popup shows just string[][][] but the line

```
$this->emailData['attachments'][$i]['message'] = base64_decode($this->emailData['attachments'][$i]['message']);
```

still shows an error.

Changed to this
```
$t = $this->emailData;
$t['attachments'][$i]['message'] = base64_decode($this->emailData['attachments'][$i]['message']);
```

and the type info of $t is array|false|string|string[][][] so it looks like PhpStorm is, for some reason, not showing its full computed type info here like it usually does.

Now that I found that I could easily find

```
$this->emailData = $this->package->PopMessageVariable();
```

and PopMessageVariable is defined as returning array|string|false.

Mystery solved.  I'll try to put together a small test case now that I know the root cause.  https://youtrack.jetbrains.com/issue/WI-41762

**Comment by kalessil** [comment]
Oh, that was found fast =)
